### PR TITLE
Send spam weird

### PR DESCRIPTION
**Pull request by javiteri95**
Add two actions: ADD_Email_label and delete_email_label

### CODE REVIEW

File: email_mailbox/src/containers/Email.js [hunk 1]
```diff
@@ -144,6 +145,7 @@ const mapDispatchToProps = (dispatch, ownProps) => {
         labelsRemoved
       })
     ).then(() => {
+      dispatch(addEmailLabel([email], LabelType.spam.id));
```
Comment (danieltigse) [inline email_mailbox/src/containers/Email.js]: move this into `action/emails.js`

File: email_mailbox/src/containers/Email.js [hunk 2]
```diff
@@ -171,6 +173,7 @@ const mapDispatchToProps = (dispatch, ownProps) => {
           labelsRemoved
         })
       ).then(() => {
+        dispatch(addEmailLabel([email], LabelType.trash.id));
```
Comment (danieltigse) [inline email_mailbox/src/containers/Email.js]: move this into `action/emails.js`

File: email_mailbox/src/actions/threads.js [hunk 3]
```diff
@@ -293,6 +309,8 @@ export const removeLabelIdThread = (currentLabelId, threadId, labelId) => {
       const dbResponse = await deleteEmailLabel(params);
       if (dbResponse) {
         dispatch(removeLabelIdThreadSuccess(currentLabelId, threadId, labelId));
+        const emails = await getEmailsByThreadId(threadId);
+        dispatch(removeEmailLabel(emails, labelId));
```
Comment (danieltigse) [inline email_mailbox/src/actions/threads.js]: Check all the cases where `SocketCommand.PEER_THREAD_LABELS_UPDATE` is used.

File: email_mailbox/src/reducers/emails.js [hunk 4]
```diff
@@ -109,6 +131,24 @@ const email = (state, action) => {
...
+      const labelIds = state.get('labelIds');
+      if (!labelIds) return state;
+      const newSet = new Set(action.labelsAdd);
+      const mergedSet = new Set([...labelIds, ...newSet]);
+      return state.set('labelIds', mergedSet);
+    }
+    case Email.DELETE_LABEL: {
+      const labelIds = state.get('labelIds');
+      if (!labelIds) return state;
+      const newSet = new Set(
```
Comment (erikaperugachi) [inline email_mailbox/src/reducers/emails.js]: to delete from SET, use .delete(value)

File: email_mailbox/src/reducers/emails.js [hunk 5]
```diff
@@ -80,6 +80,28 @@ const emails = (state = new Map(), action) => {
...
+      const emails = action.emails;
+      if (!emails || !emails.length) return state;
+      return emails.reduce((state, emailItem) => {
+        const emailId = emailItem.id;
+        if (!emailId) return state;
+        const emailState = state.get(`${emailId}`);
+        if (!emailState) return state;
+        return state.set(`${emailId}`, email(emailState, action));
+      }, state);
+    }
```
Comment (erikaperugachi) [inline email_mailbox/src/reducers/emails.js]: Delete

**Pull request merged**